### PR TITLE
Define deprecation macros conditionally

### PR DESCRIPTION
**Pull request by hkaiser**
~~- flyby: rewrite module defines.hpp only if needed~~

### CODE REVIEW

File: cmake/HPX_AddModule.cmake [hunk 1]
```diff
@@ -157,10 +157,20 @@ function(add_hpx_module libname modulename)
...
     configure_file(
       "${PROJECT_SOURCE_DIR}/cmake/templates/global_module_header.hpp.in"
-      "${global_header}"
+      "${TEMP_FILENAME}"
     )
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different "${TEMP_FILENAME}"
+              "${global_header}"
+    )
+    file(REMOVE "${TEMP_FILENAME}")
```
[the remark this answers is not in the document]
Comment (aurianer) [reply cmake/HPX_AddModule.cmake:173]: With `COPYONLY`

**Review by aurianer — COMMENTED** [review]
Thanks

File: cmake/HPX_AddModule.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (hkaiser) [reply cmake/HPX_AddModule.cmake:173]: Ok, I was not aware of that. I'm however under the impression that the files get regenerated - I will check again.

**Comment by msimberg** [comment]
> @msimberg I have removed the conditional copy operations.

And you're not getting any unnecessarily updated files anymore?

**Comment by hkaiser** [comment]
> > @msimberg I have removed the conditional copy operations.
> 
> And you're not getting any unnecessarily updated files anymore?

I have not checked, I just wanted to get the other changes in asap (need it for other projects).

**Review by msimberg — APPROVED** [review]
Thanks, looks good modulo the cmake formatting... Feel free to merge after that!

**Pull request merged**